charts/internal/cilium/charts/agent/templates/daemonset.yaml

```diff
@@ -288,6 +288,26 @@ spec:
           name: cilium-run
         resources:
           {{- toYaml .Values.monitor.resources | trim | nindent 10 }}
+{{- end }}
+      # Masquerade traffic to upstream DNS server
+{{- if .Values.global.snatToUpstreamDNS.enabled }}
+      - name: add-snat-rule-to-upstream-dns
+        image: {{ index .Values.global.images "alpine-iptables" }}
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+          capabilities:
+            drop:
+              - ALL
+            add:
+              - NET_ADMIN
+        env:
+        - name: POD_CIDR
+          value: {{ .Values.global.podCIDR }}
+        command:
+        - /bin/sh
+        - -c
+        - "while true; do for i in $(cat /etc/resolv.conf | grep nameserver | awk '{print $2}' ); do iptables -t nat -C CILIUM_POST_nat -s ${POD_CIDR} -d $i/32 ! -o cilium_+ -m comment --comment \"cilium masquerade non-cluster\" -j MASQUERADE || iptables -t nat -I CILIUM_POST_nat 1 -s ${POD_CIDR} -d $i/32 ! -o cilium_+ -m comment --comment \"cilium masquerade non-cluster\" -j MASQUERADE; done; sleep 15; done"
```

Comment:
Is it really required to run this every 15 seconds? I suppose it could run less frequently, e.g. every minute or even less.
It needs to run initially and may race with the real `cilium-agent` container, but afterwards it would only be required to be run when the `cilium-agent` container restarts, which should be sufficiently rare.